gcp-client/src/main/java/com/google/graphite/platforms/plugin/client/ComputeClient.java

```diff
@@ -0,0 +1,541 @@
+    request.setFilter(buildLabelsFilterString(labels));
+    Map<String, InstancesScopedList> result = request.execute().getItems();
+    List<Instance> instances = new ArrayList<>();
+    for (InstancesScopedList instancesInZone : result.values()) {
+      if (instancesInZone.getInstances() != null) {
+        instances.addAll(instancesInZone.getInstances());
+      }
+    }
+    return instances;
+  }
+
+  public InstanceTemplate getTemplate(String projectId, String templateName) throws IOException {
+    return compute.instanceTemplates().get(projectId, templateName).execute();
+  }
+
+  public void insertTemplate(String projectId, InstanceTemplate instanceTemplate)
+      throws IOException {
+    compute.instanceTemplates().insert(projectId, instanceTemplate).execute();
+  }
+
+  public void deleteTemplate(String projectId, String templateName) throws IOException {
+    compute.instanceTemplates().delete(projectId, templateName).execute();
+  }
+
+  public List<InstanceTemplate> getTemplates(String projectId) throws IOException {
+    List<InstanceTemplate> instanceTemplates =
+        compute.instanceTemplates().list(projectId).execute().getItems();
+    if (instanceTemplates == null) {
+      instanceTemplates = Collections.emptyList();
+    }
+
+    // Sort by name
+    instanceTemplates.sort(Comparator.comparing(InstanceTemplate::getName));
+
+    return instanceTemplates;
+  }
+
+  /**
+   * Creates persistent disk snapshot for Compute Engine instance. This method blocks until the
+   * operation completes.
+   *
+   * @param projectId Google cloud project id (e.g. my-project).
+   * @param zone Instance's zone.
+   * @param instanceId Name of the instance whose disks to take a snapshot of.
+   * @throws IOException If an error occured in snapshot creation.
+   * @throws InterruptedException If snapshot creation is interrupted.
+   */
+  public void createSnapshot(String projectId, String zone, String instanceId)
+      throws IOException, InterruptedException {
+    try {
+      zone = nameFromSelfLink(zone);
+      Instance instance = compute.instances().get(projectId, zone, instanceId).execute();
+
+      // TODO: JENKINS-56113 parallelize snapshot creation
+      for (AttachedDisk disk : instance.getDisks()) {
+        String diskId = nameFromSelfLink(disk.getSource());
+        createSnapshotForDisk(projectId, zone, diskId);
+      }
+    } catch (InterruptedException ie) {
+      // catching InterruptedException here because calling function also can throw
+      // InterruptedException from trying to terminate node
+      LOGGER.log(Level.WARNING, "Error in creating snapshot.", ie);
+      throw ie;
+    } catch (IOException ioe) {
+      LOGGER.log(Level.WARNING, "Interruption in creating snapshot.", ioe);
+      throw ioe;
+    }
+  }
+
+  /**
+   * Given a disk's name, create a snapshot for said disk.
+   *
+   * @param projectId Google cloud project id.
```

Comment:
Remove "Google cloud". That should be clear from the context.

Reply:
Done